### ISSUE TITLE
instanceof fails if checked against a variable

### ISSUE THREAD

**Issue by schinkel** [smarty-php/smarty#92]
Consider the following code:

```smarty
{$interface = 'a_random_interface'}
{$object = 'a string'}
{if $object instanceof $interface}
	<p>All good</p>
{else}
	<p>Something's wrong</p>
{/if}
```

This worked like a charm up until Smarty 3.1.23, but fails in Smarty 3.1.24 and upwards with the following error message:

```
PHP Fatal error:  Uncaught  --> Smarty Compiler: Syntax error in template "/var/www/examples/templates/correct.tpl"  on line 3 "{if $object instanceof $interface}"  - Unexpected "$interface", expected one of: "id, name" , NAMESPACE <-- 
  thrown in /var/www/examples/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatecompilerbase.php on line 3
```

I am using the following code to render this template:
```php
$smarty = new \Smarty();
$smarty
	->setCompileDir(sys_get_temp_dir())
	->setCacheDir(sys_get_temp_dir())
	->setPluginsDir(SMARTY_PLUGINS_DIR)
	->disableSecurity();
$smarty->display(__DIR__ . '/templates/correct.tpl');
```
The following snippet does compile and produces the expected "Something's wrong":

```smarty
{$interface = 'a_random_interface'}
{$object = $smarty}
{if $object instanceof aaa}
	<p>All good</p>
{else}
	<p>Something's wrong</p>
{/if}
```

**Comment by Henkiebob** [comment]
+1

**Comment by uwetews** [comment]
I can confirm this bug. The fix will go into the master branch and version 3.1.28 in a few days.

Possible fix:

in  lexer/smarty_internal_templateparser.y
```
expr(res)        ::= variable(v1) INSTANCEOF(i) ns1(v2). {
      res = v1.i.v2;
}

expr(res)        ::= variable(v1) INSTANCEOF(i) variable(v2). {
      res = v1.i.v2;
}
```
add the second expression.

Get with composer   "smarty/smarty-lexer": "~3.1@dev"
Run Create_Template_Parser.php

Because of other changes under work I can't commit it directly immediately

.

**Comment by uwetews** [comment]
The fix is now available in the master branch and will be included in the upcoming 3.1.28

**Comment by schinkel** [comment]
Cool, thanks!
